### PR TITLE
[FLINK-29274][hive] Fix ObjectStore leak when different users has dif…

### PR DESCRIPTION
**Pull request by fsk119**
…ferent config

## What is the purpose of the change

*The connection may leak when users have different config. Here, we fix this like hive does to use the same config to create the HiveMetaStoreClient.*

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2Endpoint.java [hunk 1]
```diff
@@ -296,22 +296,27 @@ public TOpenSessionResp OpenSession(TOpenSessionReq tOpenSessionReq) throws TExc
...
                     new HiveCatalog(
                             catalogName,
                             getUsedDefaultDatabase(originSessionConf).orElse(defaultDatabase),
                             conf,
                             HiveShimLoader.getHiveVersion(),
                             allowEmbedded);
+            // Trigger the creation of the HiveMetaStoreClient to use the same HiveConf. If the
+            // initial HiveConf is different, it will trigger the PersistenceManagerFactory to close
+            // all the alive PersistenceManager in the ObjectStore, which may get error like
+            // "PersistenceManager is closed" in the later connection.
```
Comment (yuzelin) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2Endpoint.java:311]: The exact exception message is “Persistence Manager has been closed”

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2Endpoint.java [hunk 2]
```diff
@@ -321,6 +326,7 @@ public TOpenSessionResp OpenSession(TOpenSessionReq tOpenSessionReq) throws TExc
                                     .setDefaultCatalog(catalogName)
                                     .addSessionConfig(sessionConfig)
                                     .build());
+            // set variables to HiveConf or Session's conf
```
Comment (yuzelin) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2Endpoint.java:329]: I think this comment line (329) should lay before line 315 0r 319.

**Review by yuzelin — APPROVED** [review]
LGTM. I left several comments, please check out.

**Comment by fsk119** [comment]
Thanks for your review. Because the comment is not releated to the code, I will this offline.

**Comment by fsk119** [comment]
The failed test is not releated to the fix. Merging...

**Pull request closed without merging**